File: SDKs/jellyfin.local/v1/python/src/sdk/models/shared/bufferrequestdto.py
from dataclasses import dataclass, field
from datetime import datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json


@dataclass_json
@dataclass
class BufferRequestDto:
    is_playing: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'IsPlaying' }})
    playlist_item_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'PlaylistItemId' }})
    position_ticks: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'PositionTicks' }})
    when: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'When', 'encoder': datetime.isoformat, 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    

File: SDKs/jellyfin.local/v1/typescript/src/sdk/models/shared/bufferrequestdto.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// BufferRequestDto
/** 
 * Class BufferRequestDto.
**/
export class BufferRequestDto extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=IsPlaying" })
  isPlaying?: boolean;

  @SpeakeasyMetadata({ data: "json, name=PlaylistItemId" })
  playlistItemId?: string;

  @SpeakeasyMetadata({ data: "json, name=PositionTicks" })
  positionTicks?: number;

  @SpeakeasyMetadata({ data: "json, name=When" })
  when?: Date;
}
